drivers/crc/crc_hw.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CRC_H
+#define ZEPHYR_INCLUDE_DRIVERS_CRC_H
+
+#include "autoconf.h"
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
The Doxygen API documentation comment is missing
```suggestion
/**
 * @brief CRC driver APIs
 * @defgroup crc_interface CRC driver APIs
 * @ingroup io_interfaces
 * @{
 */
```